youtube_dlc/extractor/cda.py

```diff
@@ -180,3 +186,50 @@ def extract_format(page, version):
         self._sort_formats(formats)
 
         return info_dict
+
+def getDecoder():
+    return js2py.eval_js(R'''
+var k = function (a) {
+    return a.replace(/[a-zA-Z]/g, function (a) {
+        return String.fromCharCode(("Z" >= a ? 90 : 122) >= (a = a.charCodeAt(0) + 13) ? a : a - 26)
+    })
+};
+
+var l = function(a) {
+    for (var b = [], e = 0; e < a.length; e++) {
+        var f = a.charCodeAt(e);
+        b[e] = 33 <= f && 126 >= f ? String.fromCharCode(33 + (f + 14) % 94) : String.fromCharCode(f)
+    }
+    return da(b.join(""))
+}
+
+var aa = function (a) {
+    String.fromCharCode(("Z" >= a ? 82 : 132) >= (c = a.charCodeAt(0) + 11) ? c : c - 55);
+    return l(a)
+};
+
+    da = function(a) {
+        a = a.replace(".cda.mp4", "");
+        a = a.replace(".2cda.pl", ".cda.pl");
+        a = a.replace(".3cda.pl", ".cda.pl");
+        return -1 < a.indexOf("/upstream") ? (a = a.replace("/upstream", ".mp4/upstream"),
+        "https://" + a) : "https://" + a + ".mp4"
+    }
+
+var ba = function (a) {
+    return aa(decodeURIComponent(k(a)))
+};
+
+// M function:
+var decode = function (a) {
+    String.fromCharCode(("Z" >= a ? 11 : 344) >= (c = a.charCodeAt(0) + 22) ? c : c - 11);
+    a = a.replace("_XDDD", "");
+    a = a.replace("_CDA", "");
+    a = a.replace("_ADC", "");
+    a = a.replace("_CXD", "");
+    a = a.replace("_QWE", "");
+    a = a.replace("_Q5", "");
+    a = a.replace("_IKSDE", "");
+    return ba(k(a))
+};
+''')
```

Comment:
this can be rewritten in pure python so there would be no need for another library (js2py).

Reply:
Good idea :)